tests/executor/test_executor.py

```diff
@@ -188,3 +198,136 @@ def test_clear(tmpfile):
         status = f.post(on='/status', return_results=True)[0]
         assert int(status.tags['total_docs']) == 0
         assert int(status.tags['index_size']) == 0
+
+
+@patch.dict(os.environ, {'JINA_AUTH_TOKEN': ''})
+def test_remote_storage(tmpfile):
+    os.environ['JINA_AUTH_TOKEN'] = '9bdfd52b39e8c32722208da4d7f4396c'
+    clear_hubble()
+
+    docs = gen_docs(N)
+    f = Flow().add(
+        uses=AnnLiteIndexer,
+        uses_with={
+            'n_dim': D,
+        },
+        workspace=tmpfile,
+        shards=1,
+    )
+    with f:
+        f.post(on='/index', inputs=docs)
+        time.sleep(2)
+        f.post(on='/backup', parameters={'target': 'backup_docs'})
+        time.sleep(2)
+
+    f = Flow().add(
+        uses=AnnLiteIndexer,
+        uses_with={'n_dim': D, 'restore_key': 'backup_docs'},
+        workspace=tmpfile,
+        shards=1,
+    )
+    with f:
+        status = f.post(on='/status', return_results=True)[0]
+
+    assert int(status.tags['total_docs']) == N
+    assert int(status.tags['index_size']) == N
+
+
+def test_local_storage(tmpfile):
+    docs = gen_docs(N)
+    f = Flow().add(
+        uses=AnnLiteIndexer,
+        uses_with={
+            'n_dim': D,
+        },
+        workspace=tmpfile,
+        shards=1,
+    )
+    with f:
+        f.post(on='/index', inputs=docs)
+        time.sleep(2)
+        f.post(on='/backup')
+        time.sleep(2)
+
+    f = Flow().add(
+        uses=AnnLiteIndexer,
+        uses_with={'n_dim': D},
+        workspace=tmpfile,
+        shards=1,
+    )
+    with f:
+        status = f.post(on='/status', return_results=True)[0]
+
+    assert int(status.tags['total_docs']) == N
+    assert int(status.tags['index_size']) == N
+
+
+@patch.dict(os.environ, {'JINA_AUTH_TOKEN': ''})
+def test_remote_storage_with_shards(tmpfile):
+    os.environ['JINA_AUTH_TOKEN'] = '9bdfd52b39e8c32722208da4d7f4396c'
+    docs = gen_docs(N)
+    f = Flow().add(
+        uses=AnnLiteIndexer,
+        uses_with={
+            'n_dim': D,
+        },
+        workspace=tmpfile,
+        shards=3,
+        polling={'/index': 'ALL', '/backup': 'ALL', '/status': 'ALL'},
```

Comment:
the pooling strategy of '/index' should be `ANY`, not `ALL`, you can refer to [this](https://docs.jina.ai/how-to/scale-out/#split-data-into-partitions-shards) to see how sharding works.

Reply:
I see

Reply:
But when change '/index' to `ANY` it failed

Reply:
I see, wait me to fix it